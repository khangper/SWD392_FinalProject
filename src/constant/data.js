export const API_KEY = "API_KEY";

export const FETCH_HOME_LIVESTREAM_REQUEST = "FETCH_HOME_LIVESTREAM_REQUEST";
export const FETCH_HOME_LIVESTREAM_SUCCESS = "FETCH_HOME_LIVESTREAM_SUCCESS";
export const FETCH_HOME_LIVESTREAM_FAILURE = "FETCH_HOME_LIVESTREAM_FAILURE";

export const FETCH_HOME_FEATURED_COURSES_REQUEST =
  "FETCH_HOME_FEATURED_COURSES_REQUEST";
export const FETCH_HOME_FEATURED_COURSES_SUCCESS =
  "FETCH_HOME_FEATURED_COURSES_SUCCESS";
export const FETCH_HOME_FEATURED_COURSES_FAILURE =
  "FETCH_HOME_FEATURED_COURSES_FAILURE";
// All Live Stream
export const FETCH_STREAMS_REQUEST = "FETCH_STREAMS_REQUEST";
export const FETCH_STREAMS_SUCCESS = "FETCH_STREAMS_SUCCESS";
export const FETCH_STREAMS_FAILURE = "FETCH_STREAMS_FAILURE";

//All Instructor
export const FETCH_INSTRUCTORS_REQUEST = "FETCH_INSTRUCTORS_REQUEST";
export const FETCH_INSTRUCTORS_SUCCESS = "FETCH_INSTRUCTORS_SUCCESS";
export const FETCH_INSTRUCTORS_FAILURE = "FETCH_INSTRUCTORS_FAILURE";
export const SEARCH_INSTRUCTORS_REQUEST = "SEARCH_INSTRUCTORS_REQUEST";
export const FETCH_INSTRUCTOR_BY_ID_REQUEST = "FETCH_INSTRUCTOR_BY_ID_REQUEST";

// SaveCourse.js
export const FETCH_SAVECOURSES_REQUEST = "FETCH_SAVECOURSES_REQUEST";
export const FETCH_SAVECOURSES_SUCCESS = "FETCH_SAVECOURSES_SUCCESS";
export const FETCH_SAVECOURSES_FAILURE = "FETCH_SAVECOURSES_FAILURE";
export const DELETE_SAVECOURSE_REQUEST = "DELETE_SAVECOURSE_REQUEST";
export const DELETE_SAVECOURSE_SUCCESS = "DELETE_SAVECOURSE_SUCCESS";
export const DELETE_SAVECOURSE_FAILURE = "DELETE_SAVECOURSE_FAILURE";

// Shoppingcart.js
export const ShoppingCart_API_URL =
  "https://66908916c0a7969efd9c67ed.mockapi.io/ojt-repo/ShoppingCart";
export const ADD_TO_CART = "ADD_TO_CART";
export const REMOVE_FROM_CART = "REMOVE_FROM_CART";
export const APPLY_DISCOUNT = "APPLY_DISCOUNT";
export const ADD_TO_CART_FAILURE = " ADD_TO_CART_FAILURE";
export const ADD_TO_CART_SUCCESS = "APPLY_DISCOUNT";

//MyCourse_Course
export const FETCH_COURSES_REQUEST = "FETCH_COURSES_REQUEST";
export const FETCH_COURSES_SUCCESS = "FETCH_COURSES_SUCCESS";
export const FETCH_COURSES_FAILURE = "FETCH_COURSES_FAILURE";

export const EDIT_COURSE_REQUEST = "EDIT_COURSE_REQUEST";
export const EDIT_COURSE_SUCCESS = "EDIT_COURSE_SUCCESS";
export const EDIT_COURSE_FAILURE = "EDIT_COURSE_FAILURE";

export const DELETE_COURSE_REQUEST = "DELETE_COURSE_REQUEST";
export const DELETE_COURSE_SUCCESS = "DELETE_COURSE_SUCCESS";
export const DELETE_COURSE_FAILURE = "DELETE_COURSE_FAILURE";

//UpcommingCourse_Course
export const FETCH_UPCOMMINGCOURSES_REQUEST = "FETCH_UPCOMMINGCOURSES_REQUEST";
export const FETCH_UPCOMMINGCOURSES_SUCCESS = "FETCH_UPCOMMINGCOURSES_SUCCESS";
export const FETCH_UPCOMMINGCOURSES_FAILURE = "FETCH_UPCOMMINGCOURSES_FAILURE";

export const EDIT_UPCOMMINGCOURSE_REQUEST = "EDIT_UPCOMMINGCOURSE_REQUEST";
export const EDIT_UPCOMMINGCOURSE_SUCCESS = "EDIT_UPCOMMINGCOURSE_SUCCESS";
export const EDIT_UPCOMMINGCOURSE_FAILURE = "EDIT_UPCOMMINGCOURSE_FAILURE";

export const DELETE_UPCOMMINGCOURSE_REQUEST = "DELETE_UPCOMMINGCOURSE_REQUEST";
export const DELETE_UPCOMMINGCOURSE_SUCCESS = "DELETE_UPCOMMINGCOURSE_SUCCESS";
export const DELETE_UPCOMMINGCOURSE_FAILURE = "DELETE_UPCOMMINGCOURSE_FAILURE";
