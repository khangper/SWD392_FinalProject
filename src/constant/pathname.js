export const PATH_NAME = {
  HOME: "/home",
  EXPLORE: "/explore",
  LIVE_OUTPUT: "/live_output/:id",
  ADD_LIVE_STREAM: "/add_livestream",
  ALL_LIVE_STREAM: "/all_livestream",
  ALL_INSTRUCTOR_LIST: "/allinstructorlist/:id",
  HELP: "/help",
  REPORT_HISTORY: "/reporthistory",
  SEND_FEEDBACK: "/sendfeedback",
  OTHER_INSTRUCTOR_VIEW: "/other-instructor-view/:id",
  MY_INSTRUCTOR_PROFILE: "/my-instructor-profile",
  SAVE_COURSES: "/save-courses",
  SETTING: "/setting",
  MY_COURSE: "/courses",
  LOGIN: "/login",
  FORGOT_PASSWORD: "/forgot-password",
  SIGN_UP: "/sign-up",
  SIGN_UP_STEP: "/SignUpStep",
  THANKS: "/thanks",
  ERROR_404: "/error404",
  INVOICE: "/invoice",
  PRESS: "/press",
  SEARCH_RESULT: "/search_result",
  PAID_MEMBERSHIP: "/paidmembership",
  ABOUT_US: "/about-us",
  OUR_BLOG: "/our-blog",
  BLOG_SINGLE_VIEW: "/blog-single-view",
  CHECKOUT: "/checkout",
  SHOPPING_CART: "/shoppingcart",
  HELP_VIEW_1: "/helpview1",
  HELP_VIEW_2: "/helpview2",
  COMPANY_DETAILS: "/company-details",
  CAREER: "/career",
  APPLY_JOB: "/apply-job",
  CONTACT_US: "/contactus",
  TERMS_OF_USE: "/termsofuse",
  CERTIFICATION_FILL_FORM: "/certificationfillform",
  CERTIFICATION_TEST: "/certificates-test",
  TEST_RESULT: "/test-result",
  MESSAGE: "/messages",
  COURSES_DETAIL_VIEW: "/courses-detail-view/:id",
  PREVIEWS: "/reviews",
  COMING_SOON: "/coming-soon",
  INS_DASHBOARD: "/ins-dashboard",
  PURCHASE_COURSE: "/PurchaseCourse",
  NOTIFICATON: "/notifications",
  DOWLOADCOURSE: "/dowloadcourse",
  CERTIFICATION_CENTER: "/certificationcenter",
  SITEMAP: "/sitemap",
  EARNING: "/earning",
  PAYOUT: "/payout",
  VERTIFICATION: "/vertification",
  MYCERTIFICATIONS: "/mycertifications",
  CREDIT: "/credits",
  STATEMENTIS: "/statementis",
  STATEMENTST: "/statementst",
  ANALYTICS: "/analytics",
  CREATE_NEW_COURSE: "/create-course",
  STD_DASHBOARD: "/student-dashboard",
  NOTIFICATIONSTUDENT: "/notificationsstudent",
};
