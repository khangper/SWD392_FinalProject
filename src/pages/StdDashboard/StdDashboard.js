import React, { useRef, useState } from "react";
import "./StdDashboard.css";
import dashboardIcon from "../../assets/dashboard.png";
import coursesIcon from "../../assets/BookHelp.png";
const StdDashboard = () => {
  const dashboardCards = [
    {
      title: "Total Purchased Courses",
      amount: "15",
      newAmount: "New 5",
      icon: "https://gambolthemes.net/html-items/cursus-new-demo/images/dashboard/online-course.svg",
      newBadgeClass: "new-sales",
    },
    {
      title: "Total Instructors Subcribing",
      amount: "45",
      newAmount: "New 3",
      icon: "https://gambolthemes.net/html-items/cursus-new-demo/images/dashboard/knowledge.svg",
      newBadgeClass: "new-enroll",
    },
  ];
  const courses = [
    {
      id: 1,
      title: "Complete Python Bootcamp: Go from zero to hero in Python 3",
      views: "1.5k",
      purchased: "150",
      likes: "1k",
      comments: "875",
      reviews: "105",
      imgSrc:
        "https://gambolthemes.net/html-items/cursus-new-demo/images/courses/img-1.jpg",
      date: "First 2 days 22 hours",
    },
    {
      id: 2,
      title: "The Complete JavaScript Course 2020: Build Real Projects!",
      views: "175k",
      purchased: "1k",
      likes: "85k",
      comments: "915",
      reviews: "255",
      imgSrc:
        "https://gambolthemes.net/html-items/cursus-new-demo/images/courses/img-2.jpg",
      date: "Second 4 days 9 hours",
    },
    {
      id: 3,
      title: "Beginning C++ Programming - From Beginner to Beyond",
      views: "17k",
      purchased: "25",
      likes: "15k",
      comments: "155",
      reviews: "15",
      imgSrc:
        "https://gambolthemes.net/html-items/cursus-new-demo/images/courses/img-3.jpg",
      date: "Third 6 days 11 hours",
    },
  ];
  const newsItems = [
    {
      id: 1,
      title: "COVID-19 Updates & Resources",
      content:
        "See the latest updates to coronavirus-related content, including changes to monetization, and access new Creator support resources.",
      icon: "https://gambolthemes.net/html-items/cursus-new-demo/images/blog/img-1.jpg",
      link: "#",
    },
    {
      id: 2,
      title: "New Features for Instructors",
      content:
        "Discover the new features available to instructors, aimed at improving the teaching and learning experience.",
      icon: "https://gambolthemes.net/html-items/cursus-new-demo/images/blog/img-2.jpg",
      link: "#",
    },
    {
      id: 3,
      title: "Community Events and Meetups",
      content:
        "Join our community events and meetups to connect with other instructors and learners.",
      icon: "https://gambolthemes.net/html-items/cursus-new-demo/images/blog/img-3.jpg",
      link: "#",
    },
  ];
  const coursePerformanceRef = useRef(null);
  const newsPerformanceRef = useRef(null);
  const scrollLeftCourse = () => {
    coursePerformanceRef.current.scrollBy({ left: -410, behavior: "smooth" });
  };
  const scrollRightCourse = () => {
    coursePerformanceRef.current.scrollBy({ left: 410, behavior: "smooth" });
  };
  const scrollLeftNews = () => {
    newsPerformanceRef.current.scrollBy({ left: -410, behavior: "smooth" });
  };
  const scrollRightNews = () => {
    newsPerformanceRef.current.scrollBy({ left: 410, behavior: "smooth" });
  };

  return (
    <div className="std-dashboard-container">
      <div className="std-dashboard-top">
        <div className="std-dashboard-header">
          <h2>
            <img src={dashboardIcon} alt="" /> Student Dashboard
          </h2>
        </div>
        <section className="dashboard-cards">
          {dashboardCards.map((card, index) => (
            <div className="dashboard-card" key={index}>
              <div className="card-content">
                <div className="card-info">
                  <h3>{card.title}</h3>
                  <h1>{card.amount}</h1>
                  <span className={`new-badge ${card.newBadgeClass}`}>
                    {card.newAmount}
                  </span>
                </div>
                <div className="card-icon">
                  <img src={card.icon} alt={`${card.title} Icon`} />
                </div>
              </div>
            </div>
          ))}
        </section>
      </div>
      <div className="std-dashboard-bottom">
        <div className="std-dashboard-left">
          <section className="course-and-news-section">
            <section className="course-performance-section">
              <div className="course-performance-header">
                <h2 className="course-performance-title">News</h2>
                <div className="course-performance-btn">
                  <button
                    className="scroll-button left"
                    onClick={() => scrollLeftNews(newsPerformanceRef)}
                  ></button>
                  <button
                    className="scroll-button right"
                    onClick={() => scrollRightNews(newsPerformanceRef)}
                  ></button>
                </div>
              </div>
              <div
                className="course-performance-container"
                ref={newsPerformanceRef}
              >
                {newsItems.map((news) => (
                  <div key={news.id} className="course-performance-card">
                    <a>
                      <img src={news.icon} alt={news.title} />
                    </a>
                    <div className="course-details">
                      <div className="course-details-header">
                        <p className="course-title">{news.title}</p>
                      </div>
                      <p className="news-content">{news.content}</p>
                      <div className="course-performance-links">
                        <a href={news.link} className="course-link">
                          Learn More
                        </a>
                      </div>
                    </div>
                  </div>
                ))}
              </div>
            </section>
            <section className="right-section">
            <section className="profile-analytics-section">
              <div className="course-performance-header">
                <h2 className="course-performance-title">
                  What's new on Cursus
                </h2>
              </div>
              <div className="sc-card">
                <a href="#" className="news-link">
                  Improved performance on Studio Dashboard
                </a>
                <a href="#" className="news-link">
                  See more Dashboard updates
                </a>
                <a href="#" className="news-link">
                  See issues-at-glance for Live
                </a>
              </div>
            </section>
          </section>
          </section>
        </div>
        <div className="std-dashboard-right">
        
        </div>
      </div>
    </div>
  );
};

export default StdDashboard;
