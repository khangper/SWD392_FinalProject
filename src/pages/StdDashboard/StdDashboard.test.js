import React from "react";
import { render, screen } from "@testing-library/react";
import StdDashboard from "../StdDashboard/StdDashboard";
import "@testing-library/jest-dom/extend-expect";
import { renderWithProviders } from "../../constant/test-util";

describe("StdDashboard", () => {
  beforeAll(() => {
    jest.spyOn(console, "error").mockImplementation(() => {});
    jest.spyOn(console, "log").mockImplementation(() => {});
  });

  afterAll(() => {
    console.error.mockRestore();
    console.log.mockRestore();
  });
  it("renders the StdDashboard component", async () => {
    const { container } = renderWithProviders(<StdDashboard />);
    expect(container).toBeDefined();
    expect(container).toMatchSnapshot();
  });
});
