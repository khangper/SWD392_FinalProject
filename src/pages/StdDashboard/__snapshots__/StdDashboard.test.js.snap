// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StdDashboard renders the StdDashboard component 1`] = `
<div>
  <div
    class="std-dashboard-container"
  >
    <div
      class="std-dashboard-top"
    >
      <div
        class="std-dashboard-header"
      >
        <h2>
          <img
            alt=""
            src="dashboard.png"
          />
           Student Dashboard
        </h2>
      </div>
      <section
        class="dashboard-cards"
      >
        <div
          class="dashboard-card"
        >
          <div
            class="card-content"
          >
            <div
              class="card-info"
            >
              <h3>
                Total Purchased Courses
              </h3>
              <h1>
                15
              </h1>
              <span
                class="new-badge new-sales"
              >
                New 5
              </span>
            </div>
            <div
              class="card-icon"
            >
              <img
                alt="Total Purchased Courses Icon"
                src="https://gambolthemes.net/html-items/cursus-new-demo/images/dashboard/online-course.svg"
              />
            </div>
          </div>
        </div>
        <div
          class="dashboard-card"
        >
          <div
            class="card-content"
          >
            <div
              class="card-info"
            >
              <h3>
                Total Instructors Subcribing
              </h3>
              <h1>
                45
              </h1>
              <span
                class="new-badge new-enroll"
              >
                New 3
              </span>
            </div>
            <div
              class="card-icon"
            >
              <img
                alt="Total Instructors Subcribing Icon"
                src="https://gambolthemes.net/html-items/cursus-new-demo/images/dashboard/knowledge.svg"
              />
            </div>
          </div>
        </div>
      </section>
    </div>
    <div
      class="std-dashboard-bottom"
    >
      <div
        class="std-dashboard-left"
      >
        <section
          class="course-and-news-section"
        >
          <section
            class="course-performance-section"
          >
            <div
              class="course-performance-header"
            >
              <h2
                class="course-performance-title"
              >
                News
              </h2>
              <div
                class="course-performance-btn"
              >
                <button
                  class="scroll-button left"
                />
                <button
                  class="scroll-button right"
                />
              </div>
            </div>
            <div
              class="course-performance-container"
            >
              <div
                class="course-performance-card"
              >
                <a>
                  <img
                    alt="COVID-19 Updates & Resources"
                    src="https://gambolthemes.net/html-items/cursus-new-demo/images/blog/img-1.jpg"
                  />
                </a>
                <div
                  class="course-details"
                >
                  <div
                    class="course-details-header"
                  >
                    <p
                      class="course-title"
                    >
                      COVID-19 Updates & Resources
                    </p>
                  </div>
                  <p
                    class="news-content"
                  >
                    See the latest updates to coronavirus-related content, including changes to monetization, and access new Creator support resources.
                  </p>
                  <div
                    class="course-performance-links"
                  >
                    <a
                      class="course-link"
                      href="#"
                    >
                      Learn More
                    </a>
                  </div>
                </div>
              </div>
              <div
                class="course-performance-card"
              >
                <a>
                  <img
                    alt="New Features for Instructors"
                    src="https://gambolthemes.net/html-items/cursus-new-demo/images/blog/img-2.jpg"
                  />
                </a>
                <div
                  class="course-details"
                >
                  <div
                    class="course-details-header"
                  >
                    <p
                      class="course-title"
                    >
                      New Features for Instructors
                    </p>
                  </div>
                  <p
                    class="news-content"
                  >
                    Discover the new features available to instructors, aimed at improving the teaching and learning experience.
                  </p>
                  <div
                    class="course-performance-links"
                  >
                    <a
                      class="course-link"
                      href="#"
                    >
                      Learn More
                    </a>
                  </div>
                </div>
              </div>
              <div
                class="course-performance-card"
              >
                <a>
                  <img
                    alt="Community Events and Meetups"
                    src="https://gambolthemes.net/html-items/cursus-new-demo/images/blog/img-3.jpg"
                  />
                </a>
                <div
                  class="course-details"
                >
                  <div
                    class="course-details-header"
                  >
                    <p
                      class="course-title"
                    >
                      Community Events and Meetups
                    </p>
                  </div>
                  <p
                    class="news-content"
                  >
                    Join our community events and meetups to connect with other instructors and learners.
                  </p>
                  <div
                    class="course-performance-links"
                  >
                    <a
                      class="course-link"
                      href="#"
                    >
                      Learn More
                    </a>
                  </div>
                </div>
              </div>
            </div>
          </section>
          <section
            class="right-section"
          >
            <section
              class="profile-analytics-section"
            >
              <div
                class="course-performance-header"
              >
                <h2
                  class="course-performance-title"
                >
                  What's new on Cursus
                </h2>
              </div>
              <div
                class="sc-card"
              >
                <a
                  class="news-link"
                  href="#"
                >
                  Improved performance on Studio Dashboard
                </a>
                <a
                  class="news-link"
                  href="#"
                >
                  See more Dashboard updates
                </a>
                <a
                  class="news-link"
                  href="#"
                >
                  See issues-at-glance for Live
                </a>
              </div>
            </section>
          </section>
        </section>
      </div>
      <div
        class="std-dashboard-right"
      />
    </div>
  </div>
</div>
`;
